src/chain.rs

```diff
@@ -18,24 +18,38 @@ use crate::{mv_memory::MvMemory, PevmTxExecutionResult};
 pub enum RewardPolicy {
     /// Ethereum
     Ethereum,
+    /// Optimism
+    #[cfg(feature = "optimism")]
+    Optimism {
+        /// L1 Fee Receipient
+        l1_fee_recipient_location_hash: crate::MemoryLocationHash,
+        /// Base Fee Vault
+        base_fee_vault_location_hash: crate::MemoryLocationHash,
+    },
 }
 
 /// Custom behaviours for different chains & networks
 pub trait PevmChain: Debug {
+    /// The transaction type
+    type Transaction: Debug + Clone + PartialEq;
```

Comment:
Let's port this, `TxEnvError`, and `get_tx_env` into a dedicated PR to merge first.

Reply:
Will be resolved in https://github.com/risechain/pevm/pull/331